x-pack/plugins/enterprise_search/public/applications/app_search/components/meta_engine_creation/meta_engine_creation.tsx

```diff
@@ -0,0 +1,175 @@
+  META_ENGINE_CREATION_FORM_MAX_SOURCE_ENGINES_WARNING_TITLE,
+  META_ENGINE_CREATION_FORM_META_ENGINE_DESCRIPTION,
+  META_ENGINE_CREATION_FORM_SUBMIT_BUTTON_LABEL,
+  META_ENGINE_CREATION_FORM_TITLE,
+  META_ENGINE_CREATION_TITLE,
+  SANITIZED_NAME_NOTE,
+} from './constants';
+import { MetaEngineCreationLogic } from './meta_engine_creation_logic';
+
+const engineNameToComboBoxOption = (engineName: string): EuiComboBoxOptionOption<string> => ({
+  label: engineName,
+});
+
+const comboBoxOptionToEngineName = (option: EuiComboBoxOptionOption<string>): string =>
+  option.label;
+
+export const MetaEngineCreation: React.FC = () => {
+  const {
+    configuredLimits: {
+      engine: { maxEnginesPerMetaEngine } = { maxEnginesPerMetaEngine: Infinity },
+    },
+  } = useValues(AppLogic);
+
+  const {
+    fetchIndexedEngineNames,
+    setRawName,
+    setSelectedIndexedEngineNames,
+    submitEngine,
+  } = useActions(MetaEngineCreationLogic);
+
+  const { rawName, name, indexedEngineNames, selectedIndexedEngineNames } = useValues(
+    MetaEngineCreationLogic
+  );
+
+  useEffect(() => {
+    fetchIndexedEngineNames();
+  }, []);
+
+  return (
+    <div data-test-subj="MetaEngineCreation">
+      <SetPageChrome trail={[META_ENGINE_CREATION_TITLE]} />
+      <EuiPageHeader>
+        <EuiPageHeaderSection>
+          <EuiTitle size="l">
+            <h1>{META_ENGINE_CREATION_TITLE}</h1>
+          </EuiTitle>
+          <EuiText>{META_ENGINE_CREATION_FORM_META_ENGINE_DESCRIPTION}</EuiText>
+          <EuiText>{META_ENGINE_CREATION_FORM_DOCUMENTATION_DESCRIPTION}</EuiText>
+        </EuiPageHeaderSection>
+      </EuiPageHeader>
+      <EuiPageBody>
+        <FlashMessages />
+        <EuiPanel>
+          <EuiForm>
+            <form
+              data-test-subj="MetaEngineCreationForm"
+              onSubmit={(e) => {
+                e.preventDefault();
+                submitEngine();
+              }}
+            >
+              <EuiTitle>
+                <EuiText>{META_ENGINE_CREATION_FORM_TITLE}</EuiText>
+              </EuiTitle>
+              <EuiSpacer />
+              <EuiFlexGroup>
+                <EuiFlexItem>
+                  <EuiFormRow
+                    data-test-subj="MetaEngineCreationNameFormRow"
+                    label={META_ENGINE_CREATION_FORM_ENGINE_NAME_LABEL}
+                    helpText={
+                      name.length > 0 && rawName !== name ? (
+                        <>
+                          {SANITIZED_NAME_NOTE} <strong>{name}</strong>
+                        </>
+                      ) : (
+                        ALLOWED_CHARS_NOTE
+                      )
+                    }
+                    fullWidth
+                  >
+                    <EuiFieldText
+                      name="engine-name"
+                      value={rawName}
+                      onChange={(event) => setRawName(event.currentTarget.value)}
+                      autoComplete="off"
```

Comment:
just curious, why the extra `autoComplete="off"` specificity here?

Reply:
Removed

Reply:
Ah sorry, I should have been more specific that it wasn't a change request, I was just curious about what the prop was doing